### PR TITLE
Uml 817 abstract out code verification

### PR DESCRIPTION
**Pull request by cooperaj**
## Purpose
Abstracts our actor one time use code usage and verification out into a strategy interface so that we can replace it with the new microservice that deals with those things

Fixes UML-817

## Approach
Creation of a new `CodeValidationStrategyInterface` and associated `DynamoCodeValidationStrategy` to migrate our existing strategy to the new interface.

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-api/app/src/App/src/Exception/ActorCodeMarkAsUsedException.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Exception;
+
```
Comment (williamfalconeruk) [inline service-api/app/src/App/src/Exception/ActorCodeMarkAsUsedException.php:6]: do these need doc blocks?

File: service-api/app/src/App/src/Exception/ActorCodeValidationException.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Exception;
+
+class ActorCodeValidationException extends \Exception
```
Comment (williamfalconeruk) [inline service-api/app/src/App/src/Exception/ActorCodeValidationException.php:7]: Do these need Doc blocks?

File: service-api/app/src/App/src/Service/Lpa/LpaService.php [hunk 3]
```diff
@@ -244,51 +240,51 @@ public function getByViewerCode(string $viewerCode, string $donorSurname, bool $
      * Given an LPA and an Actor ID, this returns the actor's details, and what type of actor they are.
      *
      * TODO: Confirm if we need to look in Trust Corporations, or if an active Trust Corporation would appear in `attorneys`.
+     * TODO: Remove dual checks for id/uId when code validation API goes live
```
Comment (williamfalconeruk) [inline service-api/app/src/App/src/Service/Lpa/LpaService.php:243]: do we need a ticket raised to cover this TODO?

Comment (cooperaj) [reply service-api/app/src/App/src/Service/Lpa/LpaService.php:243]: I've come to realise that removal of this requires the migration of all active codes to use uId. Might be easier to remove the todo :P

**Pull request merged**